src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/SourceConstantHelper.java

```diff
@@ -391,36 +396,54 @@ private String getConstantStringFieldName(String javaName) {
         return javaName + "$STR_CONSTANT_";
     }
 
-    private String emitConstantStringField(String javaName, Object value) {
+    private String emitConstantAddress(String javaName, Class<?> type, Object value) {
+        boolean isAddr = type == MemoryAddress.class;
         incrAlign();
         indent();
         String fieldName = getConstantStringFieldName(javaName);
```

Comment:
This should not be named "StringFieldName" since it now refers to both String constants and pointer constants?
Also, perhaps it would be better to just have two methods `emitXYZ` one for strings, and one for address constants, instead of having one which keeps checking `isAddr` ?